### ISSUE TITLE
Analytics setup screen loads continuously during setup - Uncaught (in promise) TypeError: Cannot read property 'replace' of undefined

### ISSUE THREAD

**Issue by jamesozzie** [google/site-kit-wp#3706]
## Bug Description

There have been 3 recent reports of the below console errors appearing, when trying to setup the Analytics module. The setup screen itself never progresses, remaining in a loading state:
`Uncaught (in promise) TypeError: Cannot read property 'replace' of undefined`

All sites using SK 1.36.0 with a first time Site Kit setup. This occurs after clicking on the "continue module setup" or when connecting Analytics for the first time via Site Kit. 

- https://wordpress.org/support/topic/analytics-not-connecting-after-new-site-uploaded/#post-14659590 - **Open** - [SH info](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit?urlBuilderDomain=google.com#response=ACYDBNjEXYowVdi8kktvInzqGnhm6HWmlTJOomRSUyDKiYnkjXUfFD2whOGdzZc0vq2pRFU)
- https://wordpress.org/support/topic/connect-analytics-in-site-kit-loads-infinitely/ - **Open** - [SH info](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit?urlBuilderDomain=google.com#response=ACYDBNjfQn1fJJBtaJwBRGYLvVGYF0cwNzXhJsIpjeTwJr7NL2PoMlz9b3IbUeBdSLL34jc)
- https://wordpress.org/support/topic/cant-activate-google-analytics/#post-14659801 - **Open** - [SH info](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit#response=ACYDBNgEHqurQt0oLegbWnxUhyIeTSxGhLzc6LqGwdAdGq7UMgWotJDDV2vxZ3x1vnPFaHo)
- https://wordpress.org/support/topic/your-site-may-not-be-ready-for-site-kit-25/ - **Open** - [SH info](https://docs.google.com/forms/d/1lhjns2K7c0Ny7ME4gxTcsnYtNUtcMr8BvFpPQi23Y1E/edit?urlBuilderDomain=get10up.com#response=ACYDBNjLM5wM5AvhCqX2mNxVby9gv7bE8JfsbSJEgm7hF3pL6HpPcJe1QmacHjOu8fTrIxk)
## Additional Context

 - All impacted sites using SK 1.36.0
 - Different hosting providers
 - Elementor is the only common plugin
 - No existing UA or GA4 snippets placed prior to SK setup

_Console errors:_

```
QMIGRATE: Migrate is installed, version 3.3.2
googlesitekit-api.c423bd9792e7eed18683.js:1 Uncaught (in promise) TypeError: Cannot read property 'replace' of undefined
    at o (googlesitekit-api.c423bd9792e7eed18683.js:1)
    at googlesitekit-modules-analytics-4.7b34aa738d17383f29ca.js:20
    at u (googlesitekit-vendor.739fd0f156bc499b3341.js:1)
    at Generator._invoke (googlesitekit-vendor.739fd0f156bc499b3341.js:1)
    at Generator.forEach.e.<computed> [as next] (googlesitekit-vendor.739fd0f156bc499b3341.js:1)
    at googlesitekit-vendor.739fd0f156bc499b3341.js:1
```

## Insights & Troubleshooting Checks performed 

- [x] Health Check & Troubleshooting plugin check - unsuccessful in both case, the issue remains
- [x] Front end check (mixed content, minified content, combined JS)
- [x] Incognito mode 
- [x] Checked users Site Health info, no identified causes. Both Linux Apache services
- [x] Additional .htaccess rules applied (1/2 sites do have additional rules)
- [x] Confirmed no modification of Site Kit placed snippet
- [ ] Uninstalled and reinstalled plugin
- [ ] User confirmed no "Critical errors" in their Site Health status
- [ ] Attempted setup/viewing dashboards from another device 
- [ ] Temporary disabled WAF rules at host level - including ModSecurity
- [ ] Checked site setup and dashboard from another WordPress site on the same hosting platform
- [x] Reset Site Kit - Issue remains
- [ ] Disconnected Site Kit
- [ ] Checked does the same occur from another admin users account
- [ ] Checked setup / viewing dashboards from another WordPress administrator account
- [ ] Reviewed error logs 
- [ ] Reviewed XHR requests 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* It should not be possible to encounter the specific error `Uncaught (in promise) TypeError: Cannot read property 'replace' of undefined` due to using the `normalizeURL` JS function with a value that's not a string.

## Implementation Brief

* Merge #3711.

### Test Coverage

* N/A

### Visual Regression Changes

* N/A

## QA Brief

* Ensure that, when passing an array of values where some of them aren't strings to the GA4 `matchPropertyByURL` action for the `url` parameter, no JS error occurs, and only the string values will be considered.

## Changelog entry

* Fix potential `Uncaught (in promise) TypeError: Cannot read property 'replace' of undefined` related to GA4 property matching logic.

**Comment by jamesozzie** [comment]
Possible third case, awaiting details: https://wordpress.org/support/topic/cant-activate-google-analytics/#post-14659801

**Comment by tofumatt** [comment]
QA ✅

**Comment by BusfyJan** [comment]
I too have this problem (continuous loading + error in console)

![image](https://user-images.githubusercontent.com/5645935/125834494-f14d4348-4202-4028-bc8f-d1341abfbcae.png)
![image](https://user-images.githubusercontent.com/5645935/125834452-b9b0d79c-7636-41bf-8122-bda2dce58380.png)

**Comment by abdullah1908** [comment]
@BusfyJan ,

Thanks for reported the issue. Our team has fixed this issue & hopefully you'll see this fix in upcoming Site Kit release `1.37.0`.

Have a great weekend 😀
